scipy/optimize/meson.build

```diff
@@ -159,17 +159,23 @@ __nnls = py3.extension_module('__nnls',
   install: true,
   subdir: 'scipy/optimize')
 
-res = run_command(py3, '-c', 'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 1))')
-if res.returncode() == 0
+res = run_command(py3, '-c', 'import os; print(os.environ.get("SCIPY_USE_PYTHRAN", 1))').stdout().strip()
+if res == '1'
   _group_columns = custom_target('_group_columns',
     output: ['_group_columns.cpp'],
     input : '_group_columns.py',
     command : [pythran, '-E', '@INPUT@', '-o', '@OUTDIR@/_group_columns.cpp'],
   )
   _group_columns = py3.extension_module('_group_columns',
   [_group_columns],
-  cpp_args: ['-Wno-unused-function', '-Wno-int-in-bool-context'],
-  include_directories: [inc_pythran],
+  cpp_args: ['-Wno-unused-function', '-Wno-unused-variable', 
+             '-Wno-deprecated-declarations',
+             '-Wno-cpp', '-Wno-int-in-bool-context', 
+             '-DENABLE_PYTHON_MODULE',
+             '-D__PYTHRAN__=3',
+             '-DPYTHRAN_BLAS_NONE'],
```

Comment:
Let me know if there is a robust way to do this. I just picked these flags from the following output of `setup.py`,

```python
compile options: '-DENABLE_PYTHON_MODULE -D__PYTHRAN__=3 -DPYTHRAN_BLAS_NONE -I/home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/lib/python3.9/site-packages/pythran -I/home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/lib/python3.9/site-packages/numpy/core/include -I/home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/lib/python3.9/site-packages/numpy/core/include -Ibuild/src.linux-x86_64-3.9/numpy/distutils/include -I/home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/include/python3.9 -I/home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/include/python3.9 -c'
extra options: '-std=c++11 -fno-math-errno -w -fvisibility=hidden -fno-wrapv -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/czgdp1807ssd/anaconda3/envs/scipy-dev-main/include -msse -msse2 -msse3'
x86_64-conda-linux-gnu-c++: scipy/optimize/_group_columns.cpp
```

Reply:
Interesting, that wasn't necessary for other Pythran extensions (I believe; maybe tests for those other extensions are failing and I just haven't seen that yet) - I'll think about the best way of doing this.

Reply:
> (I believe; maybe tests for those other extensions are failing and I just haven't seen that yet) 

Yeah, it's only during tests that `ImportError` is being thrown. I checked, `build.ninja`  these flags were missing there. 

Ah! I see. In CPP file generated by Pythran, the above flags are used. For example,

```cpp
#include <pythonic/python/exception_handler.hpp>
#ifdef ENABLE_PYTHON_MODULE
.
.
.
#endif
```

All other code is inside this block. Hence, if we don't pass this flag, export function is never defined. It is optional so build never fails. We need these flags as it is may be because anyways, these are not platform dependent but required by Pythran.

Reply:
Adding the flags like that looks good. I pushed one commit to move them to `scipy/meson.build` and give them a name (`cpp_args_pythran`) so it can be reused in other submodules.